crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id),
+                redirect: false,
+                redirection_data: None,
+                mandate_reference: None,
+            }),
+            amount_captured: None,
+            ..item.data
+        })
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Eq, PartialEq, Default, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum OrderStatus {
+    New,
+    Canceled,
+    Completed,
+    #[serde(rename = "WAITING_FOR_CONFIRMATION")]
+    WaitingForConfirmation,
+    #[default]
+    Pending,
+}
+
+impl From<OrderStatus> for enums::AttemptStatus {
+    fn from(item: OrderStatus) -> Self {
+        match item {
+            OrderStatus::New => Self::PaymentMethodAwaited,
+            OrderStatus::Canceled => Self::Voided,
+            OrderStatus::Completed => Self::Charged,
+            OrderStatus::Pending => Self::Pending,
+            OrderStatus::WaitingForConfirmation => Self::Authorized,
+        }
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Serialize, Default, Deserialize, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuPaymentStatusData {
+    status_code: PayuPaymentStatus,
+    severity: Option<String>,
+    status_desc: Option<String>,
+}
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuProductData {
+    name: String,
+    unit_price: String,
+    quantity: String,
+    #[serde(rename = "virtual")]
+    virtually: Option<bool>,
+    listing_date: Option<String>,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuOrderResponseData {
+    order_id: String,
+    ext_order_id: Option<String>,
+    order_create_date: String,
+    notify_url: Option<String>,
+    customer_ip: String,
+    merchant_pos_id: String,
+    description: String,
+    validity_time: Option<String>,
+    currency_code: String,
+    total_amount: String,
+    buyer: Option<PayuOrderResponseBuyerData>,
+    pay_method: Option<PayuOrderResponsePayMethod>,
+    products: Option<Vec<PayuProductData>>,
+    status: OrderStatus,
+}
+
+#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct PayuOrderResponseBuyerData {
+    ext_customer_id: Option<String>,
+    email: Option<String>,
+    phone: Option<String>,
+    first_name: Option<String>,
+    last_name: Option<String>,
+    nin: Option<String>,
+    language: Option<String>,
+    delivery: Option<String>,
+    customer_id: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "type")]
+#[serde(rename_all = "UPPERCASE")]
+pub enum PayuOrderResponsePayMethod {
+    #[serde(rename = "CARD_TOKEN")]
+    CardToken,
+    Pbl,
+    Installemnts,
+}
```

Comment:
You can use `#[serde(rename_all = "SCREAMING_SNAKE_CASE")] here.

Also, you can just merge the `#[serde]` annotations: `#[serde(tag = "...", rename_all = "...")]`